### DISCUSSION TITLE
sFlow L2 support

### DISCUSSION THREAD

**Discussion by MarianRychtecky** [akvorado/akvorado#657] [Q&A]
Hi all,
I'm interested in support of sFlow L2 headers such as Destination and Source MAC in metrics (dimensions). Is this feature considered? Ideally, with data enrichment from either a static config file or database. Thanks for your response.

**Comment by vincentbernat** [comment]
You can enable source and destination MAC address here: https://demo.akvorado.net/docs/configuration#schema

**Reply by MarianRychtecky** [reply]
Hi Vincent, thank you for the hint. I'm confused about what configuration file I have to put this in. Please help me understand which file and what section configuration schema belongs. Thanks

**Reply by MarianRychtecky** [reply]
Please disregard it; I found it in akvorado.conf in the main section. Thank you

**Comment by MarianRychtecky** [comment]
Hi again,
 I'm trying to understand how to enrich data sFlow based on dstMAC and srcMAC. I want to use InIfProvider and OutIfProvider fields.

Is there any better way how to do it instead of trying to modify it in here:

`  interface-classifiers:
    - |
      ClassifyConnectivityRegex(Interface.Description, "^.*(H|P|T|S|)>", "$1") &&
      ClassifyProviderRegex(Interface.Description, "^\\S+?\\s(\\S+)", "$1") &&
`
And what would be the correct ClassifyProviderRegex parameter for dst/src MAC? I looked at the documentation, and MACs are missing.

Thank you, Marian

**Reply by vincentbernat** [reply]
It is currently not possible.

**Reply by MarianRychtecky** [reply]
No problem; thank you for your response. Are there any plans to make it possible?

**Comment by MarianRychtecky** [comment]
Hi all,
I would like to ask you for a hint. We are ingesting sFlow just fine; the graphs are statistically correct and shown properly. The only issue now is the volume of calculated traffic is way higher than SNMP counter readings. For instance, SNMP graphs (and CLI on the interface) show ~250Gbit/s, but the sFlow graph in Akvorado shows ~390Gbit/s. I believe this has something to do with packet size or so. I haven't found any recommendation for sFlow settings other than the sampling ratio. I will appreciate, if you can point me in the right direction.
